src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -120,4 +122,29 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
             intent,
         };
     }
+
+    private _getFindingKey(methodId: string): keyof PaypalCommerceSDKFunding {
```

Comment:
A suggestion. Instead of injecting an enum type into the class and check against it to determine the `fundingKey`, perhaps it's better to inject an object that can directly return the `fundingKey` value based on `methodId/gatewayId`? i.e.: You can have something like 

```ts
class PaypalCommerceFundingKeyResolver {
    resolve(methodId, gatewayId) {
        if (methodId === 'paypalcommerce') {
            return 'PAYPAL';
        }

        if (methodId === 'paypalcommercecredit') {
            return 'PAYLATER';
        }

        if (methodId === 'bancontact' && gatewayId === 'paypalcommercealternativemethods') {
            return 'BANCONTACT';
        }

        // Etc...

        throw new Error('Unable to resolve funding key');
    }
}
```

And then in `PaypalCommercePaymentStrategy`, you can just ask for the value by calling `PaypalCommerceFundingKeyResolver#resolve`? The benefit is that you don't need to introduce another/secondary "strategy type" to provide branching behaviours. Also it's easier to test as you can test the resolver on its own. Thoughts?

Reply:
Thank you @davidchin !
I added function `paypalCommerceFundingKeyResolver` for this